packages/passport/sdk-sample-app/src/components/zkevm/EthSendTransactionExamples/SeaportFulfillAvailableAdvancedOrders.tsx

```diff
@@ -0,0 +1,172 @@
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import {
+  Accordion, Form, Spinner, Stack,
+} from 'react-bootstrap';
+import WorkflowButton from '@/components/WorkflowButton';
+import { RequestExampleProps } from '@/types';
+import { useImmutableProvider } from '@/context/ImmutableProvider';
+import { usePassportProvider } from '@/context/PassportProvider';
+import { ActionType, TransactionAction, TransactionPurpose } from '@imtbl/orderbook';
+
+function SeaportFulfillAvailableAdvancedOrders({ disabled, handleExampleSubmitted }: RequestExampleProps) {
+  const { orderbookClient } = useImmutableProvider();
+  const { zkEvmProvider } = usePassportProvider();
+
+  const [listingIds, setListingIds] = useState<string>('');
+  const [walletAddress, setWalletAddress] = useState<string>('');
+  const [isBuldingTransaction, setIsBuildingTransaction] = useState<boolean>(false);
+  const [transaction, setTransaction] = useState<PopulatedTransaction>();
+  const [transactionError, setTransactionError] = useState<string>('');
+
+  const seaportContractAddress = useMemo(
+    () => (
+      orderbookClient.config().seaportContractAddress
+    ),
+    [orderbookClient],
+  );
+
+  useEffect(() => {
+    const getAddress = async () => {
+      if (zkEvmProvider) {
+        const [address] = await zkEvmProvider.request({
+          method: 'eth_requestAccounts',
+        });
+        setWalletAddress(address || '');
+      }
+    };
+
+    getAddress().catch(console.log);
+  }, [zkEvmProvider, setWalletAddress]);
+
+  useEffect(() => {
+    setTransactionError('');
+    setTransaction(undefined);
+  }, [listingIds]);
+
+  const validate = useCallback(async () => {
+    setTransactionError('');
+    setTransaction(undefined);
+    setIsBuildingTransaction(true);
+
+    try {
+      const fulfillResponse = await orderbookClient.fulfillBulkOrders(
+        listingIds.replace(' ', '').split(',').map((orderId) => ({
+          listingId: orderId,
+          takerFees: [],
+        })),
+        walletAddress,
+      );
+
+      if (!fulfillResponse.sufficientBalance) {
+        setTransactionError('Insufficient balance to fulfill the order(s)');
+        return;
+      }
+
+      if (fulfillResponse.sufficientBalance) {
+        const {
+          actions, unfulfillableOrders,
+        } = fulfillResponse;
+
+        if (unfulfillableOrders.length > 0) {
+          setTransactionError('Not all orders are fulfillable');
+          return;
+        }
+
+        const verifyAction = actions.find((action) => (
+          action.type === ActionType.TRANSACTION && action.purpose === TransactionPurpose.APPROVAL
+        )) as TransactionAction | undefined;
+
+        if (verifyAction) {
+          const approvalTransaction = await verifyAction.buildTransaction();
+          await zkEvmProvider?.request({
+            method: 'eth_requestAccounts',
+            params: [approvalTransaction],
+          });
+        }
```

Comment:
q: what's that for?

Reply:
This is the approval transaction for the ERC20 spending cap. If the buy transaction is for an ERC20 and the user does not have sufficient spending cap allowances the orderbook will return an approval transaction to sign.

Reply:
KK, why calling `eth_requestAccounts` can grant the approval?